kratos/containers/set_function.h

```diff
@@ -0,0 +1,115 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#pragma once
+
+// System includes
+#include <functional>
+#include <type_traits>
+#include <utility>
+
+// External includes
+
+// Project includes
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/**
+ * @class TestOperators
+ * @brief Operator class to test whether operators "<" and "==" is defined for given TDataType.
+ */
+struct TestOperators
+{
+    template<class TDataType>
+    static auto TestEqual(TDataType*) -> decltype(std::declval<TDataType>() == std::declval<TDataType>());
+    template<typename>
+    static auto TestEqual(...) -> std::false_type;
+
+    template<class TDataType>
+    static auto TestLess(TDataType*) -> decltype(std::declval<TDataType>() < std::declval<TDataType>());
+    template<typename>
+    static auto TestLess(...) -> std::false_type;
+};
+
+/**
+ * @class SetFunction
+ * @brief A functor that serves as the identity function if "<" and "==" operators are defined, otherwise, returns memory location.
+ * @details This class provides two overloaded operator() functions, one for non-const objects
+ * and another for const objects. The operator() returns the input objects memory location as it is,
+ * effectively acting as an identity function for address if the operators "<" and "==" is not defined. Otherwise,
+ * this returns the exact input object.
+ *
+ * The purpose of this functor is to allow objects' memory locations or exact object to be used as keys in sets or other
+ * containers that require comparison. By using this functor, you can avoid defining
+ * custom comparison functions or operators when the object itself can be considered
+ * for comparison.
+ * @tparam TDataType The data type of the object that the functor operates on.
+ * @author Suneth Warnakulasuriya
+ */
+template<class TDataType>
+class SetFunction
```

Comment:
Would you please give a more descriptive name to this function? `PointerSetGetKey`?

Reply:
done :)... Changed to `KeyGenerator`.